cpp/tests/test_abm.cpp

```diff
@@ -193,30 +193,30 @@ TEST(TestLocation, interact)
     //test should work identically work with any age
     epi::AbmAgeGroup age = epi::AbmAgeGroup(epi::UniformIntDistribution<int>()(0, int(epi::AbmAgeGroup::Count) - 1));
     epi::GlobalInfectionParameters params;
-    params.set<epi::CarrierToInfected>({{epi::AbmAgeGroup::Count}, 0.});
-    params.get<epi::CarrierToInfected>()[{age}] = 0.5;
-    params.set<epi::CarrierToRecovered>({{epi::AbmAgeGroup::Count}, 0.});
-    params.get<epi::CarrierToRecovered>()[{age}] = 0.5;
-    params.set<epi::DetectInfection>({{epi::AbmAgeGroup::Count}, 0.});
-    params.get<epi::DetectInfection>()[{age}] = 0.5;
-    params.set<epi::InfectedToSevere>({{epi::AbmAgeGroup::Count}, 0.});
-    params.get<epi::InfectedToSevere>()[{age}] = 0.5;
-    params.set<epi::InfectedToRecovered>({{epi::AbmAgeGroup::Count}, 0.});
-    params.get<epi::InfectedToRecovered>()[{age}] = 0.5;
-    params.set<epi::SevereToCritical>({{epi::AbmAgeGroup::Count}, 0.});
-    params.get<epi::SevereToCritical>()[{age}] = 0.5;
-    params.set<epi::SevereToRecovered>({{epi::AbmAgeGroup::Count}, 0.});
-    params.get<epi::SevereToRecovered>()[{age}] = 0.5;
-    params.set<epi::CriticalToDead>({{epi::AbmAgeGroup::Count}, 0.});
-    params.get<epi::CriticalToDead>()[{age}] = 0.5;
-    params.set<epi::CriticalToRecovered>({{epi::AbmAgeGroup::Count}, 0.});
-    params.get<epi::CriticalToRecovered>()[{age}] = 0.5;
-    params.set<epi::RecoveredToSusceptible>({{epi::AbmAgeGroup::Count}, 0.});
-    params.get<epi::RecoveredToSusceptible>()[{age}] = 0.5;
-    params.set<epi::SusceptibleToExposedByCarrier>({{epi::AbmAgeGroup::Count}, 0.});
-    params.get<epi::SusceptibleToExposedByCarrier>()[{age}] = 0.5;
-    params.set<epi::SusceptibleToExposedByInfected>({{epi::AbmAgeGroup::Count}, 0.});
-    params.get<epi::SusceptibleToExposedByInfected>()[{age}] = 0.5;
+    params.set<epi::CarrierToInfected>({{epi::AbmAgeGroup::Count,epi::VaccinationState::Count}, 0.});
```

Comment:
please randomize the vaccination state in the same way as the age to make sure the vaccination state is used correctly in the infection transitions.

Reply:
Done, every person beneath had to be changed to the vaccination state which was picked randomly.